### PR TITLE
NMS-7475: Changed NodeAvailabilityReport.jrxml to include open outages

### PR DESCRIPTION
**Pull request by soleger**
Include open outages in the raw outage calculation.

* JIRA: http://issues.opennms.org/browse/NMS-7475
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS795

### CODE REVIEW

File: opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml [hunk 1]
```diff
@@ -46,15 +46,24 @@
 		</parameter>
 		<parameter name="DS_AVAILABILITY_WARNING" class="java.lang.Double" isForPrompting="false"/>
 		<parameter name="DS_AVAILABILITY_CRITICAL" class="java.lang.Double" isForPrompting="false"/>
+		<parameter name="TIME_RANGE_NUMBER" class="java.lang.Integer" isForPrompting="false">
```
Comment (mvrueden) [inline opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml]: This should be `DS_TIME_RANGE_NUMBER`.

File: opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml [hunk 2]
```diff
@@ -46,15 +46,24 @@
...
 		<parameter name="DS_TIME_RANGE" class="java.lang.String" isForPrompting="false">
-			<defaultValueExpression><![CDATA["7 Days"]]></defaultValueExpression>
+			<defaultValueExpression><![CDATA[$P{TIME_RANGE_NUMBER} + " Days"]]></defaultValueExpression>
 		</parameter>
 		<parameter name="DS_START_TIME" class="java.util.Date">
-			<defaultValueExpression><![CDATA[new java.util.Date(new java.util.Date().getTime() - (86400 * 7 * 1000))]]></defaultValueExpression>
+			<defaultValueExpression><![CDATA[new java.util.Date(new java.util.Date().getTime() - (86400000L * $P{TIME_RANGE_NUMBER}))]]></defaultValueExpression>
 		</parameter>
 		<parameter name="DS_START_TIME_STRING" class="java.lang.String" isForPrompting="false">
 			<defaultValueExpression><![CDATA["yyyy-MM-d HH:mm:ss Z"]]></defaultValueExpression>
```
Comment (soleger) [inline opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml]: I have a question: shouldn't this defaultValueExpression be like this instead? Right now, it's just the DateFormat string which doesn't seem right.
```
<![CDATA[new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format($P{DS_START_TIME})]]>
```

Comment (mvrueden) [inline opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml]: Yes, I think so too.

**Comment by soleger** [comment]
Changed merged into foundation-2016, closing PR.

**Pull request closed without merging**